sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -161,84 +200,204 @@ private boolean methodIsIneligible(MethodVertex vertex) {
                 || PathEntryPointUtil.isPathEntryPoint(vertex);
     }
 
-    private boolean directedToSkip(MethodVertex methodVertex) {
-        List<EngineDirective> directives = methodVertex.getAllEngineDirectives();
-        for (EngineDirective directive : directives) {
-            if (directive.isAnyDisable()
-                    && directive.matchesRule(this.getClass().getSimpleName())) {
-                return true;
-            }
-        }
-        return false;
-    }
-
+    /**
+     * Seeks invocations of the provided method that occur within the class where it's defined.
+     * E.g., `this.method()` or `method()`, etc.
+     *
+     * @return - True if such an invocation can be found, else false.
+     */
     private boolean methodUsedInternally(MethodVertex methodVertex) {
         // Instantiate a visitor to use for checking internal calls.
         InternalCallValidator visitor = new InternalCallValidator(methodVertex);
         // Get every method call in the class containing the method.
         List<InvocableWithParametersVertex> potentialInternalCallers =
-                getInternalCalls(methodVertex.getDefiningType());
+                getMethodCalls(methodVertex.getDefiningType());
+        return validatorDetectsUsage(visitor, potentialInternalCallers);
+    }
 
-        // For each internal call...
-        for (InvocableWithParametersVertex potentialCall : potentialInternalCallers) {
-            // Use our visitor to determine whether this invocation is of the target method.
-            if (!visitor.isInternalCall(potentialCall)) {
-                continue;
+    /**
+     * @return True if this provided method is visible to subclasses of the class where it's
+     *     defined.
+     */
+    private boolean methodVisibleToSubclasses(MethodVertex methodVertex) {
```

Comment:
Looks like we are trying to see if a method is extensible in a subclass beyond only visibility, since a protected method would be visible but not extensible - can this be renamed accordingly?

Reply:
Actually, we're only looking for visibility. Whether the method is overridden is relevant later, but not here.